oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/search/ShouldActiveFiltersRender.jsx

```diff
@@ -14,8 +14,10 @@ const ShouldActiveFiltersRenderComponent = ({
     initialQueryState: { filters: initialFilters },
   } = searchAppContext;
   return (
-    <ShouldRender condition={filters.length > initialFilters?.length}>
-      {children}
+    <ShouldRender condition={filters?.length > initialFilters?.length}>
+      {React.Children.map(children, (child) =>
+        React.cloneElement(child, { filters, initialFilters })
```

Comment:
This smells like a hack :) Can't children components (the ones that actually needs it?) also directly access `currentQueryState` & `initialQueryState` through context provider ?

Reply:
@mirekys I fully expected a comment here:) The problem is, initialQueryState can indeed be accessed from the context, but currentQueryState only via withState HOC. So now I have this annoying situation, where my ShouldActiveFilterRender already is wrapped with withState, and now, as its child in this case also needs it, I need to wrap it in withState again to access currentQueryState and this seemed like an antipattern i.e. they don't, make the searchkit state accessible in a very convenient way. If this is a big issue, I think I could replace this children and cloning with a render prop, where it will be something like this

`   <ShouldRender condition={filters?.length > initialFilters?.length}>
      {/* {React.Children.map(children, (child) =>
        React.cloneElement(child, { filters, initialFilters })
      )} */}
      {children({ filters, initialFilters })}
    </ShouldRender>`
    
    And then I need to replace code everywhere where I am using this component to
    
    `   <ShouldActiveFiltersRender>
              {() => (
                <ClearFiltersButton
                  className={"clear-filters-button mobile tablet only"}
                />
              )}
            </ShouldActiveFiltersRender>`
            
            To be honest, I find the current implementation nicer to use, but please let me know how you would prefer for it to be done. Thanks!